gradle/build-logic/convention/src/main/kotlin/AndroidLibraryPublishingConventionPlugin.kt

```diff
@@ -12,6 +13,13 @@ class AndroidLibraryPublishingConventionPlugin : Plugin<Project> {
       apply("dev.drewhamilton.poko")
     }
 
+    // exclude poko annotations dependency from runtime as it's only needed for compilation
+    configurations.all {
+      if (name.endsWith("RuntimeClasspath")) {
+        exclude(group = "dev.drewhamilton.poko")
+      }
+    }
```

Comment:
Interesting. Can you help me understand why this is needed? Does it have to do with the way I'm applying `poko` above? If not, should all projects that use `poko` do this?

Reply:
the gradle plugin is adding the annotation dependency using `implementation` instead of `compileOnly`. It's seems that there's been some back-and-forth on trying to change that which apparently broke something https://github.com/drewhamilton/Poko/issues/163.

Looking at the [implementation](https://github.com/drewhamilton/Poko/blob/dc4ff0288959705886a3eb09d748325d9f43a0d2/poko-gradle-plugin/src/main/kotlin/dev/drewhamilton/poko/gradle/PokoGradlePlugin.kt#L20) an alternative would be to use a custom annotation and use `compileOnly` yourself, which would get rid of the extra pom metadata, but that seems like a lot to set up for something that should be pretty simple imo.

Reply:
👋 It was switched back to `implementation` in drewhamilton/poko#212 because native targets apparently don't have the concept of `compileOnly`. If Telephoto is Android-only, this should be safe.

If you want to file an issue on Poko to make non-native targets use `compileOnly` by default (again), I wouldn't hate it!

Reply:
Ah, it's not Android-only. In that case you'll want to avoid applying this change to native targets.

Reply:
Interesting, yeah I'm using this in an android project which is where I noticed the issue

Reply:
so I checked and while this lib is multiplatform the only other platform is desktop jvm, which I would expect to behave the same way in terms of compilation right? I was going to submit a pr porting the sample app to desktop to prove this out but I couldn't get it working, 😕.

Reply:
Yea I only have desktop and android, but I want to add support for targets once coil becomes stable. I forgot to ask, what's the downside of not merging this?